src/main/java/sg/ncl/MainController.java

```diff
@@ -3494,6 +3486,514 @@ public String adminEnergy(Model model,
+        }
+
+        return "redirect:/admin/monthly";
+    }
+
+    @GetMapping("/admin/monthly/{id}/usage")
+    public String adminMonthlyUsage(@PathVariable String id, HttpSession session, Model model) {
+        if (!validateIfAdmin(session)) {
+            return NO_PERMISSION_PAGE;
+        }
+
+        HttpEntity<String> request = createHttpEntityHeaderOnly();
+        ResponseEntity response = restTemplate.exchange(properties.getMonthly() + "/" + id, HttpMethod.GET, request, String.class);
+        JSONObject jsonObject = new JSONObject(response.getBody().toString());
+        ProjectDetails projectDetails = getProjectDetails(jsonObject);
+        model.addAttribute("project", projectDetails);
+
+        return "admin_monthly_usage";
+    }
+
+    @GetMapping(value = {"/admin/monthly/{id}/usage/contribute", "/admin/monthly/{id}/usage/contribute/{month}"})
+    public String adminMonthlyUsageContribute(@PathVariable String id, @PathVariable Optional<String> month,
+                                              HttpSession session, Model model) {
+        if (!validateIfAdmin(session)) {
+            return NO_PERMISSION_PAGE;
+        }
+
+        if (month.isPresent()) {
+            HttpEntity<String> request = createHttpEntityHeaderOnly();
+            ResponseEntity response = restTemplate.exchange(properties.getMonthlyUsage(id) + "/" + month.get(), HttpMethod.GET, request, String.class);
+            JSONObject usage = new JSONObject(response.getBody().toString());
+            JSONObject usageId = usage.getJSONObject("id");
+            ProjectUsage projectUsage = new ProjectUsage();
+            projectUsage.setId(usageId.getInt("projectDetailsId"));
+            projectUsage.setMonth(usageId.getString("monthYear"));
+            projectUsage.setUsage(usage.getInt("monthlyUsage"));
+            model.addAttribute("usage", projectUsage);
+        } else {
+            model.addAttribute("usage", new ProjectUsage());
+        }
+        model.addAttribute("pid", id);
+
+        return "admin_monthly_usage_contribute";
+    }
+
+    @PostMapping("/admin/monthly/{pid}/usage/contribute")
+    public String adminMonthlyUsageValidate(@Valid @ModelAttribute("usage") ProjectUsage usage, BindingResult binding,
+                                            @PathVariable String pid, HttpSession session, Model model) throws WebServiceRuntimeException {
+        if (!validateIfAdmin(session)) {
+            return NO_PERMISSION_PAGE;
+        }
+
+        if (binding.hasErrors()) {
+            StringBuilder message = new StringBuilder();
+            message.append("Error(s):");
+            message.append("<ul class=\"fa-ul\">");
+            for (ObjectError objectError : binding.getAllErrors()) {
+                FieldError fieldError = (FieldError) objectError;
+                message.append("<li><i class=\"fa fa-exclamation-circle\"></i> ");
+                switch (fieldError.getField()) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this "switch" statement by "if" statements to increase readability. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1301)